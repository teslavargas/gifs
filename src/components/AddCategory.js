import React, { useState } from 'react'
import PropTypes from 'prop-types'; 

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState(''); 
    const handdleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const handdleSubmit = (e) => {
        e.preventDefault(); 
        
        if(inputValue.trim().length > 2){
            setCategories(cats => [inputValue, ...cats]);
            setInputValue(''); 
        }
    
    }

    return (
        <form onSubmit={handdleSubmit}>
            <p> { inputValue } </p>
            <input type="text" value={inputValue} onChange={handdleInputChange}/>
        </form>

    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}