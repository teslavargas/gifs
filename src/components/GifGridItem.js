import React from 'react'
import PropTypes from 'prop-types';

export const GifGridItem = ({title, url}) => {

    return (
        <div className="card animate__animated animate__rubberBand">
            <img src={url} alt={title}/>
            <p className="titulo">{title}</p> 
        </div>
    )
}

GifGridItem.protoTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired

}