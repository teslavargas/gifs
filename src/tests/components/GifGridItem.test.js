import React from 'react'; 
import { shallow } from 'enzyme'; 
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en GifGridItem', () => {

    const titulo = 'Un título';
    const url = 'https://localhost/algo.png'; 
    const wrapper = shallow(<GifGridItem title={titulo} url={url}/>)

    test('Debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot(); 
    })

    test('Debe tener un párrafo con el título', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(titulo); 
    })
    
    test('Debe tener la imagen igual al url y alt de los props', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(titulo); 

    })
    
    test('Debe tener animate__rubberBand', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('animate__rubberBand')).toBe(true);  
    })
    
    
})
