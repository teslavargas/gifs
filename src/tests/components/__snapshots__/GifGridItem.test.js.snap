// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__rubberBand"
>
  <img
    alt="Un título"
    src="https://localhost/algo.png"
  />
  <p
    className="titulo"
  >
    Un título
  </p>
</div>
`;
